dataflow/spanner-io/src/main/java/com/example/dataflow/TransactionalRead.java

```diff
@@ -101,27 +112,20 @@ static class Album {
 
   public static void main(String[] args) {
     Options options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
-    Pipeline p = Pipeline.create(options);
+    Pipeline pipeline = Pipeline.create(options);
 
     String instanceId = options.getInstanceId();
     String databaseId = options.getDatabaseId();
+    Dialect dialect = options.getDialect();
 
-    // [START spanner_dataflow_txread]
-    SpannerConfig spannerConfig = SpannerConfig.create()
-        .withInstanceId(instanceId)
-        .withDatabaseId(databaseId);
-    PCollectionView<Transaction> tx = p.apply(
-        SpannerIO.createTransaction()
-            .withSpannerConfig(spannerConfig)
-            .withTimestampBound(TimestampBound.strong()));
-    PCollection<Struct> singers = p.apply(SpannerIO.read()
-        .withSpannerConfig(spannerConfig)
-        .withQuery("SELECT SingerID, FirstName, LastName FROM Singers")
-        .withTransaction(tx));
-    PCollection<Struct> albums = p.apply(SpannerIO.read().withSpannerConfig(spannerConfig)
-        .withQuery("SELECT SingerId, AlbumId, AlbumTitle FROM Albums")
-        .withTransaction(tx));
-    // [END spanner_dataflow_txread]
+    Tuple<PCollection<Struct>, PCollection<Struct>> records;
+    if (dialect == Dialect.POSTGRESQL) {
+      records = pgRead(instanceId, databaseId, pipeline);
+    } else {
+      records = spannerRead(instanceId, databaseId, pipeline);
+    }
```

Comment:
Same suggestion as before for the naming
- googleSqlRead
- postgresSqlRead

Reply:
Done